jina/peapods/pods/__init__.py

```diff
@@ -509,21 +445,17 @@ def update_pea_args(self):
 
     def update_worker_pea_args(self):
         """ Update args of all its worker peas based on Pod args. Does not touch head and tail"""
-        self.peas_args['peas'] = self._set_peas_args(
-            self.args,
-            head_args=self.peas_args['head'],
-            tail_args=self.peas_args['tail'],
-        )
+        self.peas_args['peas'] = self._set_peas_args(self.args)
 
     @property
     def first_pea_args(self) -> Namespace:
-        """Return the first non-head/tail pea's args
+        """Return the first worker pea's args
 
 
         .. # noqa: DAR201
         """
         # note this will be never out of boundary
-        return self.peas_args['peas'][0]
+        return self.peas_args['peas'][0][0]
```

Comment:
why does it have depth of 2, this dict should contain `peas`, `head`, `uses_before`, `uses_after`?

Reply:
Ah is because u have `shards`

Reply:
Is this property needed? should we try to remove?

Reply:
you mean `self.peas_args`? This is used everywhere, in particular to store the shard/replica information. At the moment we need it, we could refactor like moving to a distinct class or so, but we will need sth like this.